dns/exported/exported.go

```diff
@@ -0,0 +1,19 @@
+package exported
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Entry is the entry used in the DNS.
+type Entry interface {
+	GetOwners() []sdk.AccAddress
+	GetContent() []byte
+	GetSequence() uint64
```

Comment:
Hmm.. will each update have to be precommitted before committing? ...I suppose so, though maybe there's a cleaner way

Reply:
The attack model we are trying to prevent is someone frontrunning your content registration by registering the same content bytes under a different domain

Reply:
If we do this, then precommits should be stored in a different way. Just store the latest precommit in a seperate part of the store. If it ever gets a valid commit entry, then we delete the precommit.

Only have a single precommit at any given time.

Reply:
so another issue with the current mapping is preventing double registration of content, because we use prefix + contentName there is no way to check if content exists already atm since the contentHash is not being used in any mapping, but idk how important this would be to a DNS

Reply:
> Just store the latest precommit in a seperate part of the store. If it ever gets a valid commit entry, then we delete the precommit.
So just prefix reserve two prefixes, 1 for pre-commits and 1 for content hash -> domain owners?

Reply:
Hmm, couldn't this occur:
- I send a pre-commit
- I send in a commit msg
- validator intercepts and gets content hash
- validator blocks commit msg and submits their own pre-commit using a different random nonce
- validator submits commit revealing their random nonce and registers content under their own name

Reply:
> so another issue with the current mapping is preventing double registration of content, because we use prefix + contentName there is no way to check if content exists already atm since the contentHash is not being used in any mapping, but idk how important this would be to a DNS

Don't think that's an issue. We can have many aliases to the same content hash

> I send a pre-commit
> I send in a commit msg
> validator intercepts and gets content hash
> validator blocks commit msg and submits their own pre-commit using a different random nonce
> validator submits commit revealing their random nonce and registers content under their own name

No because the commit msg must come from same signer as corresponding pre-commit